### PR TITLE
Reduce memory required for Fourier transform.

### LINKED ISSUE #94: Memory usage reduction
compute_fourier_basis uses self._e, self._U = np.linalg.eigh(self.L.toarray()) this could be replaced with scipy.linalg.eigh(self.L.toarray(), overwrite_a=True) to reuse the memory used by the output of self.L.toarray().

### PR DESCRIPTION
**Pull request by pulquero**
Fixes #94 .

### CODE REVIEW

File: pygsp/graphs/fourier.py [hunk 1]
```diff
@@ -171,7 +171,7 @@ def compute_fourier_basis(self, n_eigenvectors=None):
 
         # TODO: handle non-symmetric Laplacians. Test lap_type?
         if n_eigenvectors == self.n_vertices:
-            self._e, self._U = np.linalg.eigh(self.L.toarray())
+            self._e, self._U = linalg.eigh(self.L.toarray(order='F'), overwrite_a=True)
```
Comment (pulquero) [inline pygsp/graphs/fourier.py:174]: order='F' as lapack is fortran ordered, and re-use space for eigenvectors.

Comment (mdeff) [reply pygsp/graphs/fourier.py:174]: Thanks! Does the Fortran order make the EVD faster?

Comment (pulquero) [reply pygsp/graphs/fourier.py:174]: It should avoid the matrix being transposed from C order to F order when it is pushed out to lapack.

**Comment by mdeff** [comment]
Rebased to fix CI.

File: pygsp/graphs/fourier.py (hunk 1 above)
Comment (mdeff) [reply pygsp/graphs/fourier.py:174]: Makes sense, thanks. At some point we should do a proper benchmark and find the best settings for the partial and full EVDs.

**Comment by mdeff** [comment]
Thanks for the improvement!

**Pull request merged**